lib/pluginmanager/command.rb

```diff
@@ -1,8 +1,16 @@
 class LogStash::PluginManager::Command < Clamp::Command
+
+  def initialize(invocation_path, context = {}, parent_attribute_values = {})
+    initialize_gemfile
+    super(invocation_path, context, parent_attribute_values)
+  end
+
```

Comment:
ok 2 things here:
- first, setting `Gem.sources` does not belong in the `Command` class. if you look at how the plugin manager is architectured you will see that there's no direct interaction with the `Gem` module, everything is decoupled through the `LogStash::Bundler` and `LogStash::Rubygems` modules (and in the pluginmanager utils.rb). Second, I do not like the idea of a globally setting `Gem.sources`, this is a setting that should be kept close to the calls that require it so that it is explicit. it will be easier to refactor in the future and/or support options like setting specific sources for specific calls for example.

- refactoring the lazy initialization stuff does not belong in this PR, if we want to revisit why/how this is done here, let's create another PR, AFAIK this is not a problem right now.